File: clients/typescript/src/SettingsReply.ts
import {SettingsFormat} from './SettingsFormat';
import {SettingsFile} from './SettingsFile';
import {GitCommit} from './GitCommit';
export class SettingsReply {
  private _format?: SettingsFormat;
  private _filename?: SettingsFile;
  private _content?: string;
  private _headGitCommit?: string;
  private _gitHistory?: GitCommit[];

  constructor(input: {
    format?: SettingsFormat,
    filename?: SettingsFile,
    content?: string,
    headGitCommit?: string,
    gitHistory?: GitCommit[],
  }) {
    this._format = input.format;
    this._filename = input.filename;
    this._content = input.content;
    this._headGitCommit = input.headGitCommit;
    this._gitHistory = input.gitHistory;
  }

  get format(): SettingsFormat | undefined { return this._format; }
  set format(format: SettingsFormat | undefined) { this._format = format; }

  get filename(): SettingsFile | undefined { return this._filename; }
  set filename(filename: SettingsFile | undefined) { this._filename = filename; }

  get content(): string | undefined { return this._content; }
  set content(content: string | undefined) { this._content = content; }

  get headGitCommit(): string | undefined { return this._headGitCommit; }
  set headGitCommit(headGitCommit: string | undefined) { this._headGitCommit = headGitCommit; }

  get gitHistory(): GitCommit[] | undefined { return this._gitHistory; }
  set gitHistory(gitHistory: GitCommit[] | undefined) { this._gitHistory = gitHistory; }
}

File: clients/rust/src/settings_reply.rs
// SettingsReply represents a SettingsReply model.
#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct SettingsReply {
    #[serde(rename="format")]
    format: Box<crate::SettingsFormat>,
    #[serde(rename="filename")]
    filename: Box<crate::SettingsFile>,
    #[serde(rename="content")]
    content: String,
    #[serde(rename="head_git_commit")]
    head_git_commit: String,
    #[serde(rename="git_history")]
    git_history: Vec<crate::GitCommit>,
}

impl SettingsReply {
    pub fn new(format: crate::SettingsFormat, filename: crate::SettingsFile, content: String, head_git_commit: String, git_history: Vec<crate::GitCommit>) -> SettingsReply {
        SettingsReply {
            format: Box::new(format),
            filename: Box::new(filename),
            content,
            head_git_commit,
            git_history,
        }
    }
}
